lnwallet/channel.go

```diff
@@ -1566,6 +1571,19 @@ func (lc *LightningChannel) logUpdateToPayDesc(logUpdate *channeldb.LogUpdate,
 			ShaOnionBlob:             wireMsg.ShaOnionBlob,
 			removeCommitHeightRemote: commitHeight,
 		}
+
+	// For fee updates we'll create a FeeUpdate type to add to the log. We
+	// reuse the amount field to hold the fee rate. Since the amount field
+	// is denominated in msat we won't lose precision when storing the
+	// sat/kw denominated feerate.
+	case *lnwire.UpdateFee:
+		pd = &PaymentDescriptor{
```

Comment:
`addCommitHeightRemote` isn't set. We need to properly record this in order to handle the retransmission case.

Reply:
```
--- FAIL: TestChannelRetransmissionFeeUpdate (0.13s)
    channel_test.go:3998: unable to complete bob's state transition: rejected commitment: commit_height=2, invalid_commit_sig=30440220682b6dd35e569da6cbe26568e5388bae4c76a9ed61162087bc77471c4558ae9102207a8dad8ffd4a8297c8f50463c2c8a1177ded151f2ea003ee8bedf83f20bdfc1b, commit_tx=0200000001b794385f2d1ef7ab4d9273d1906381b44f2f6f2588a3efb96a491883319847530000000000d1af1c8003204e0000000000002200204e62432fb93d883e0910926ed09e70556f2ca6d62fa74096d016fc389ec55cdbe016cd1d00000000220020d7ba6761e1ac17cbb0722c5818bb61d312ea660cf3505f0eeef4b39f81a2d4330065cd1d000000001600144e8187664a78afec5b8c6991dbc52d575ef5e27116539e20, sig_hash=9ffec9eac9de4aaeb680da3c95bb650468f48a2b57f7ef7eac81caa5bd6863ee
```

Reply:
You're right. Test failure was due to https://github.com/lightningnetwork/lnd/commit/516a6a794cb8b2048c51a2430b9e8784de2d2d45, which is fixed in https://github.com/lightningnetwork/lnd/pull/2397/commits/821cb23f529651cf3202ff58beaf3577a364cebe